### ISSUE TITLE
Implement scalar list defaults proposal in query engine and client

### ISSUE THREAD

**Issue by tomhoule** [prisma/prisma#13321]
https://www.notion.so/prismaio/Proposal-Scalar-List-Defaults-94d0891a3355475999f10b270e2a311a

TL;DR we will allow `@default(...)` attributes on scalar list fields in PSL. The behaviour should match that of regular scalar defaults.

Models like the following are currently not valid, but they will become valid on connectors that support scalar lists (mongodb, cockroachdb, postgres):

```prisma
model Pizza {
  id       BigInt   @id @default(autoincrement())
  toppings String[] @default(["cheese", "tomato sauce", "pineapple"])
}
```

On first approximation, the foreseeable client work to make this work is:

- Make the query engine insert these defaults when no value for the field is provided for the user, for consistency with other scalar defaults.
- Determine if this needs to be reflected in the client API. The PSL proposal says "The client will not need to change since it was already possible to not specify values for scalar lists or provide undefined when writing client requests.".

**Comment by tomhoule** [comment]
https://github.com/prisma/prisma-engines/pull/2946 was merged — feel free to close this if you consider the feature done from the client / QE side.
